lib/puppet/parser/scope.rb

```diff
@@ -460,6 +460,17 @@ def lookupdefaults(type)
     values
   end
 
+  # Check if the given value is a known default for the given type
+  #
+  def is_default?(type, key, value)
+    defaults_for_type = @defaults[type]
+    unless defaults_for_type.nil?
+      default_param = defaults_for_type[key]
+      return true if !default_param.nil? && value == default_param.value
```

Comment:
Yes. You will be able to add an override as long as the value is the default value. Keeping track of if an override to the exact same value just to prevent that another override later changes it (or doesn't) didn't seem worth the added complexity.